packages/near-membrane-base/src/membrane.ts

```diff
@@ -250,15 +246,27 @@ export function init(
     // to the callable functions used to coordinate work between the sides
     // of the membrane.
     // TODO: do we need to pass more info into instrumentation hooks?
-    function instrumentCallableWrapper<T extends (...args: any[]) => any>(fn: T, label: string): T {
+    // prettier-ignore
+    function instrumentCallableWrapper<T extends (...args: any[]) => any>(
+        fn: T,
+        activityName: string,
+        crossingDirection: string): T {
         if (instrumentation) {
             return <T>function instrumentedFn(this: any, ...args: any[]): any {
                 let result;
-                instrumentation.start(label);
+                const activity = instrumentation.startActivity(activityName, { crossingDirection });
+                let error: Error;
                 try {
                     result = ReflectApply(fn, this, args);
+                } catch (ex) {
+                    if (ex instanceof Error) {
```

Comment:
I doubt that this is really needed, and if the argument passed into `activity.error` is problematic, then it should be there where the checks occur rather than happening on every place where we do the error.

Reply:
OK. Removed the type checks. Leave it for the instrumentation wrapper to worry about it.